video and ml design doc (#43)
Capstone-Skynet/Capstone-Skynet.github.io: docs/design/3_1_1_video_capture.tex

% TODO: Talk about camera choice, MIPI interface, camera libraries. Also, talk about alternatives we explored (i.e. analog video transmission/acquisition.

> The video capture subsystem of the computing platform consists of a camera, its video transmission protocol to the processing core and video transmission to the base station. For this subsystem, the hardware used includes a CMOS camera, a Rasberry Pi device, and MIPI Camera Serial Interface protocol. Design description of the Rasberry Pi device is described in the next section.

Reviewer: ```suggestion
The video capture subsystem of the computing platform consists of a camera, its video transmission protocol to the processing core and video transmission to the base station. For this subsystem, the hardware used includes a CMOS camera, a Raspberry Pi device, and MIPI Camera Serial Interface protocol. Design description of the Raspberry Pi device is described in the next section.
```

\subsubsection{Camera Choice}

> The camera used to capture video is the Rasberry Pi Camera Module. It supports 1080p, 720p and 640x480p video. It has a fixed focus lens. The camera will capture live video and send the video data to the Rasberry Pi, which will feed the data in different forms to both the machine learning model and the base station.

Reviewer: ```suggestion
The camera used to capture video is the Raspberry Pi Camera Module. It supports 1080p, 720p and 480p video. It has a fixed focus lens. The camera will capture live video and send the video data to the Raspberry Pi, which will feed the data in different forms to both the machine learning model and the base station.
```
Hardware shouldn't include protocol.

% camera Choice

> The decision of the camera was influenced by the requirement of future replacability. The project's goal is to create a system which provides both video acquisition and supplemental programmable hardware processing capabilities. The Rasberry Pi device contains different ports that will support different cameras with different performance.

Reviewer: ```suggestion
The decision of the camera was influenced by the requirement of future replaceability. The project's goal is to create a system which provides both video acquisition and supplemental programmable hardware processing capabilities. The Raspberry Pi contains different ports that will support different cameras with different performance.
```

> The decision of the camera was influenced by the requirement of future replaceability. The project's goal is to create a system which provides both video acquisition and supplemental programmable hardware processing capabilities. The Raspberry Pi device contains different ports that will support different cameras with different performance.

Reviewer 2: ```suggestion
The decision of the camera was influenced by the requirement of future replaceability. The project's goal is to create a system which provides both video acquisition and supplemental programmable hardware processing capabilities. The Raspberry Pi device contains different ports which support different cameras with varying characteristics and performance.
```

> One of the limitations on the resolution and framerate of the camera is the latency of the machine learning implementation on the FPGA. The machine learning implementation is not the focus of this project and will not be optimized for performance and thus it will not be necessary to use a camera with high resoltuion and framerate.

Reviewer: ```suggestion
One of the limitations on the resolution and framerate of the camera is the latency of the machine learning implementation on the FPGA. The machine learning implementation is not the focus of this project and will not be optimized for performance and thus it will not be necessary to use a camera with high resolution and framerate.
```
This sounds like the limitation on the throughput of the computing platform, not the camera.

> One of the limitations on the resolution and framerate of the camera is the latency of the machine learning implementation on the FPGA. The machine learning implementation is not the focus of this project and will not be optimized for performance and thus it will not be necessary to use a camera with high resolution and framerate.

Reviewer 2: ```suggestion
One major consideration regarding selecting a camera is the latency and throughput of the machine learning implementation on the FPGA. As the machine learning implementation will be heavily constrained by the FPGA's limited resources, it is not necessary to use a camera with an especially high resolution and framerate.
```
We shouldn't say that we're not optimizing things for performance :)

% {MIPI interface}
\subsubsection{MIPI Interface}
The MIPI camera serial interface (CSI) is a high-speed protocol primarily intended for point-to-point image and video transmission between cameras and host devices. 

> In the Requirements Specification, the camera is required to use a standard interface to allow for future replaceability. The MIPI protocol is a widely adopted protocol for image and video transmission. A hardware based MIPI core will be programmed into the programmable logic and will handle receiving of camera footage.

Reviewer: We are not using a MIPI core on FPGA, at least not at this point.

> In the Requirements Specification, the camera is required to use a standard interface to allow for future replaceability. The MIPI protocol is a widely adopted protocol for image and video transmission. A hardware based MIPI core will be programmed into the programmable logic and will handle receiving of camera footage.

Reviewer 2: ```suggestion
The MIPI protocol is a widely adopted protocol for image and video transmission, fulfilling requirement (XX). 
```
Maybe talk about why we didn't use something like a USB webcam?

% {Camera libraries or getting the video to a 'viewable' state}
\subsubsection{Camera Libraries}

> Live video feed is a requirement for the base station. This is achieved by implementing a Python script on the Rasberry Pi for streaming the video to a webpage on the base station via WiFi. Justification for using WiFi is discussed in Section 3.1.5 - Communications. The script used to achieve this uses the Python package 'picamera,' which is a Python interface to the Rasberry Pi Camera. 

Reviewer: ```suggestion
Live video feed is a requirement for the base station. This is achieved by implementing a Python script on the Raspberry Pi for streaming the video to a webpage on the base station via Wi-Fi. Justification for using Wi-Fi is discussed in Section 3.1.5 - Communications. The script used to achieve this uses the Python package 'picamera', which is a Python library to interface with the camera. 
```

> Live video feed is a requirement for the base station. This is achieved by implementing a Python script on the Raspberry Pi for streaming the video to a webpage on the base station via Wi-FI. Justification for using Wi-FI is discussed in Section 3.1.5 - Communications. The script used to achieve this uses the Python package 'picamera,' which is a Python interface to the Raspberry Pi Camera. 

Reviewer 2: Use the \ref command to link sections, rather than fully writing them out (just in case we move them around).

Also, why do we use the picamera library? Perhaps because it is the most widely supported library for this camera?

% {Explored Alternatives}
\subsubsection{Explored Alternatives}
There were many alternatives explored for each component of the Video Capture subsystem.

> For the choice of the camera, one of the explored possibilites was a drone package with an integrated camera. The potential difficulties of having to disassemble and reconfigure/tap-into a hard-wired system would create an unmaintainable product. It would also go against the requirement of a replaceable camera.

Reviewer 2: ```suggestion
For the choice of the camera, one of the explored possibilities was purchasing a drone pre-integrated with an camera. The implicit requirement to reconfigure/tap-into a hard-wired system, however, would create an unmaintainable product -- not allowing for user replacement of the camera for future research improvements (violating constraint XX).
```

> For the live video feed transmission, other options explored include wireless HDMI, VGA and analog transmission methods. Again, detailed justification for using WiFi and not these is discussed in a later section but the decision came down to a few factors. The original options required additional hardware (higher costs and system weight) and would require more complexity for the user to handle.

Reviewer: May not be necessary.

> For the live video feed transmission, other options explored include wireless HDMI, VGA and analog transmission methods. Again, detailed justification for using Wi-FI and not these is discussed in a later section but the decision came down to a few factors. The original options required additional hardware (higher costs and system weight) and would require more complexity for the user to handle.

Reviewer 2: Maybe talk about the additional licenses required?
